src/Data/Fin/Subset/Properties.agda

```diff
@@ -203,6 +256,171 @@ p⊆q⇒∣p∣<∣q∣ {p = outside ∷ p} {inside  ∷ q} p⊆q = ≤-step (p
 p⊆q⇒∣p∣<∣q∣ {p = inside  ∷ p} {outside ∷ q} p⊆q = contradiction (p⊆q here) λ()
 p⊆q⇒∣p∣<∣q∣ {p = inside  ∷ p} {inside  ∷ q} p⊆q = s≤s (p⊆q⇒∣p∣<∣q∣ (drop-∷-⊆ p⊆q))
 
+
+------------------------------------------------------------------------
+-- _⊂_
+
+p⊂q⇒p⊆q : ∀ {n} → {p q : Subset n} → p ⊂ q → p ⊆ q
+p⊂q⇒p⊆q = proj₁
+
+⊂-trans : ∀ {n} → Transitive (_⊂_ {n})
+⊂-trans (p⊆q , x , x∈q , x∉p) (q⊆r , _ , _ , _) = ⊆-trans p⊆q q⊆r , x , q⊆r x∈q , x∉p
+
+⊂-trans-⊆ : ∀ {n} → Trans (_⊂_ {n}) (_⊆_ {n}) (_⊂_ {n})
+⊂-trans-⊆ (p⊆q , x , x∈q , x∉p) q⊆r = ⊆-trans p⊆q q⊆r , x , q⊆r x∈q , x∉p
+
+⊆-trans-⊂ : ∀ {n} → Trans (_⊆_ {n}) (_⊂_ {n}) (_⊂_ {n})
+⊆-trans-⊂ p⊆q (q⊆r , x , x∈r , x∉q) = ⊆-trans p⊆q q⊆r , x , x∈r , λ x∈p → contradiction (p⊆q x∈p) x∉q
+
+⊂-irref : ∀ {n} → Irreflexive _≡_ (_⊂_ {n})
+⊂-irref refl (_ , x , x∈p , x∉q) = contradiction x∈p x∉q
+
+⊂-antisym : ∀ {n} → Antisymmetric _≡_ (_⊂_ {n})
+⊂-antisym (p⊆q , _) (q⊆p , _) = ⊆-antisym p⊆q q⊆p
+
+⊂-asymmetric : ∀ {n} → Asymmetric (_⊂_ {n})
+⊂-asymmetric (p⊆q , _) (_ , x , x∈p , x∉q) = contradiction (p⊆q x∈p) x∉q
+
+infix 4 _⊂?_
+
+_⊂?_ : ∀ {n} → B.Decidable (_⊂_ {n = n})
+[]          ⊂? []          = no λ ()
+outside ∷ p ⊂? outside ∷ q = Dec.map out⊂out-⇔ (p ⊂? q)
+outside ∷ p ⊂? inside  ∷ q = Dec.map out⊂in-⇔ (p ⊆? q)
+inside  ∷ p ⊂? outside ∷ q = no (λ {(p⊆q , _) → case (p⊆q here) of λ ()})
+inside  ∷ p ⊂? inside  ∷ q = Dec.map in⊂in-⇔ (p ⊂? q)
+
+module _ (n : ℕ) where
+
+  ⊂-isStrictPartialOrder : IsStrictPartialOrder _≡_ (_⊂_ {n})
+  ⊂-isStrictPartialOrder = record
+    { isEquivalence = isEquivalence
+    ; irrefl = ⊂-irref
+    ; trans = ⊂-trans
+    ; <-resp-≈ = (λ {refl → id}) , (λ {refl → id})
+    }
+
+  ⊂-strictPartialOrder : StrictPartialOrder _ _ _
+  ⊂-strictPartialOrder = record
+    { _≈_ = _≡_
+    ; _<_ = _⊂_
```

Comment:
We usually omit these first two lines entirely from the record as it's nearly always obvious what the underlying relations are. Agda will then infer them for us.

Reply:
Done.